server/pkg/api/accountHandlers.go

```diff
@@ -38,8 +43,19 @@ func createAccount(ctx apiContext, w http.ResponseWriter, r *http.Request) {
 
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error: %s\n", err.Error())
-		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-		return
+		if err.Error() != "" {
+			profile := Profile{err.Error()}
+			js, err := json.Marshal(profile)
```

Comment:
The json package can actually do all of this for you! You can replace everything from 48-53 with:

`json.NewEncoder(w).Encode(profile)` or something very similar.